cli/smartcontract/generate_test.go

```diff
@@ -0,0 +1,349 @@
+    getFromMap.indices: "[]string"
+    getFromMap: "[]int"
+    getBlockWrapper: int # check argument override
+    myFunc.in: "map[int]github.com/heyitsme/mycontract.Input"
+    myFunc: "[]github.com/heyitsme/mycontract.Output"
+callflags:
+    doSomething: ReadStates
+`
+		cfgPath := filepath.Join(t.TempDir(), "binding.yml")
+		require.NoError(t, ioutil.WriteFile(cfgPath, []byte(rawCfg), os.ModePerm))
+
+		require.NoError(t, app.Run([]string{"", "generate-wrapper",
+			"--manifest", manifestFile,
+			"--config", cfgPath,
+			"--out", outFile,
+			"--hash", h.StringLE(),
+			"--override", "getBlockWrapper=ledger.Block",
+		}))
+
+		data, err := ioutil.ReadFile(outFile)
+		require.NoError(t, err)
+		require.Equal(t, expected, string(data))
+	})
+}
+
+func TestGenerateValidPackageName(t *testing.T) {
+	m := manifest.NewManifest("My space\tcontract")
+	m.ABI.Methods = append(m.ABI.Methods,
+		manifest.Method{
+			Name:       "get",
+			Parameters: []manifest.Parameter{},
+			ReturnType: smartcontract.IntegerType,
+		},
+	)
+
+	manifestFile := filepath.Join(t.TempDir(), "manifest.json")
+	outFile := filepath.Join(t.TempDir(), "out.go")
+
+	rawManifest, err := json.Marshal(m)
+	require.NoError(t, err)
+	require.NoError(t, ioutil.WriteFile(manifestFile, rawManifest, os.ModePerm))
+
+	h := util.Uint160{
+		0x04, 0x08, 0x15, 0x16, 0x23, 0x42, 0x43, 0x44, 0x00, 0x01,
+		0xCA, 0xFE, 0xBA, 0xBE, 0xDE, 0xAD, 0xBE, 0xEF, 0x03, 0x04,
+	}
+	app := cli.NewApp()
+	app.Commands = []cli.Command{generateWrapperCmd}
+	require.NoError(t, app.Run([]string{"", "generate-wrapper",
+		"--manifest", manifestFile,
+		"--out", outFile,
+		"--hash", h.StringLE(),
+	}))
+
+	data, err := ioutil.ReadFile(outFile)
+	require.NoError(t, err)
+	require.Equal(t, `// Package myspacecontract contains wrappers for My space	contract contract.
+package myspacecontract
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop/contract"
+	"github.com/nspcc-dev/neo-go/pkg/interop/neogointernal"
+)
+
+// Hash contains contract hash in big-endian form.
+const Hash = "\x04\x08\x15\x16\x23\x42\x43\x44\x00\x01\xca\xfe\xba\xbe\xde\xad\xbe\xef\x03\x04"
+
+// Get invokes `+"`get`"+` method of contract.
+func Get() int {
+	return neogointernal.CallWithToken(Hash, "get", int(contract.All)).(int)
+}
+`, string(data))
+}
+
+func TestGenerate_Errors(t *testing.T) {
+	app := cli.NewApp()
+	app.Commands = []cli.Command{generateWrapperCmd}
+	app.ExitErrHandler = func(*cli.Context, error) {}
+
+	checkError := func(t *testing.T, msg string, args ...string) {
+		// cli.ExitError doesn't implement wraping properly, so we check for an error message.
+		err := app.Run(append([]string{"", "generate-wrapper"}, args...))
+		require.True(t, strings.Contains(err.Error(), msg), "got: %v", err)
+	}
+	t.Run("missing manifest argument", func(t *testing.T) {
+		checkError(t, errNoManifestFile.Error())
+	})
+	t.Run("missing manifest file", func(t *testing.T) {
+		checkError(t, "no such file", "--manifest", "notexists")
```

Comment:
Please, pay attention to the Windows test jobs, it was fixed in the last week and now it should pass. Currently, it fails with the following error:
```
2022-02-11T17:11:09.9030604Z === RUN   TestGenerate_Errors/missing_manifest_file
2022-02-11T17:11:09.9031316Z     generate_test.go:311: 
2022-02-11T17:11:09.9031987Z         	Error Trace:	generate_test.go:311
2022-02-11T17:11:09.9046434Z         	            				generate_test.go:317
2022-02-11T17:11:09.9049374Z         	Error:      	Should be true
2022-02-11T17:11:09.9100543Z         	Test:       	TestGenerate_Errors/missing_manifest_file
2022-02-11T17:11:09.9103511Z         	Messages:   	got: can't read contract manifest: open notexists: The system cannot find the file specified.
```